src/app/[locale]/(simulation)/(large-layout)/amis/resultats/_components/UpdateSimulationUsed.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import TransClient from '@/components/translation/trans/TransClient'
+import Button from '@/design-system/inputs/Button'
+import Card from '@/design-system/layout/Card'
+import Loader from '@/design-system/layout/Loader'
+import { formatCarbonFootprint } from '@/helpers/formatters/formatCarbonFootprint'
+import { displaySuccessToast } from '@/helpers/toasts/displaySuccessToast'
+import { useClientTranslation } from '@/hooks/useClientTranslation'
+import { useUser } from '@/publicodes-state'
+import { updateGroupParticipant } from '@/services/groups/updateGroupParticipant'
+import type { Group } from '@/types/groups'
+import { captureException } from '@sentry/nextjs'
+import dayjs from 'dayjs'
+import { useState } from 'react'
+
+type Props = {
+  group: Group
+  refetchGroup: () => void
+}
+
+export default function UpdateSimulationUsed({ group, refetchGroup }: Props) {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const {
+    user: { userId, name, email },
+    simulations,
+  } = useUser()
+
+  const { t } = useClientTranslation()
+
+  // The user hasn't got newer simulation than the one used in the group
+  if (
+    group.participants.some(
+      (participant) =>
+        participant.userId === userId &&
+        !simulations.some(
+          (simulation) =>
+            simulation.progression === 1 &&
+            dayjs(simulation.date).isAfter(dayjs(participant.simulation.date))
+        )
+    )
+  ) {
+    return null
+  }
+
+  const latestSimulation = simulations
+    .filter(
+      (simulation) =>
+        simulation.progression === 1 &&
+        dayjs(simulation.date).isAfter(
+          dayjs(
+            group.participants.find(
+              (participant) => participant.userId === userId
+            )?.simulation.date
+          )
+        )
+    )
+    .sort((a, b) => dayjs(b.date).diff(dayjs(a.date)))[0]
```

Comment:
```suggestion
  const groupSimulation = group.participants.find((p) => p.userId === userId)
  
  const newSimulation = simulations.filter((s) => s.progression === 1 && dayjs(s.date).isAfter(dayjs(groupSimulation?.date))).sort((a, b) => dayjs(b.date).diff(dayjs(a.date))).shift()
  
  if (!newSimulation) {
    return null
  }
```

2 loops O(n) + sort on filtererd list instead of 2 loops O(n2) + sort on the whole list

Reply:
top !